app/controllers/application_controller.rb

```diff
@@ -42,11 +42,18 @@ def default_url_options(_options = {})
   end
 
   def page_not_found
-    render 'shared/http_status',
-           formats: [:html],
-           locals: { code: '404', message: HttpStatusHelper::ERROR_CODE['message']['404'] },
-           status: :not_found,
-           layout: false
+    if current_user
+      if current_user.admin_user?
```

Comment:
You don't need to replicate the implementation of `current_role` here. You can just call the method so that it assigns to `@current_role`. But also it seems like this should only be called if `current_course` isn't nil, so you can do something like `current_role unless current_course.nil?`